File: solucoes/2016/nivel jr/fase 2/js/fuga.js
var H, P, F, D 

scanf("%d%d%d%d", "H", "P", "F", "D")

var pos = F

if(D === -1){
    while(True){
        pos = (pos - 1) % 16
        printf(pos)
        if(pos === H){
            printf("S")
            break
        }
        if(pos === P){
            printf("%s","N")
            break
        }
    }
}
else{  
    while(True){
        pos = (pos + 1) % 16
        if(pos === H){
            printf("%s", "S")
            break
        }
        if(pos === P){
            printf("%s", "N")
            break
        }
    }
}

File: solucoes/2016/nivel jr/fase 2/py/fuga.py
H, P, F, D = list(map(int, input().split()))

pos = F

if D == -1: 
    while True:
        pos = (pos - 1) % 16
        print(pos)
        if pos == H:
            print("S")
            break
        if pos == P:
            print("N")
            break
else:  
    while True:
        pos = (pos + 1) % 16
        if pos == H:
            print("S")
            break
        if pos == P:
            print("N")
            break
